apps/browser/src/platform/browser/browser-api.ts

```diff
@@ -184,7 +184,21 @@ export class BrowserApi {
     name: string,
     callback: (message: any, sender: chrome.runtime.MessageSender, response: any) => unknown
   ) {
-    chrome.runtime.onMessage.addListener(callback);
+    chrome.runtime.onMessage.addListener(
+      (msg: any, sender: chrome.runtime.MessageSender, sendResponse: any) => {
+        if (
+          msg.command === "fido2RegisterCredentialRequest" ||
+          msg.command === "fido2GetCredentialRequest"
+        ) {
+          // The callback needs to return a truthy value here for fido2 requests so that we can return a response
+          // using sendResponse. This can sometimes return non-boolean values which are considered truthy,
+          // so we need to explicitly check for true
+          return callback(msg, sender, sendResponse) === true;
+        } else {
+          callback(msg, sender, sendResponse);
+        }
+      }
+    );
```

Comment:
```suggestion
  chrome.runtime.onMessage.addListener(
    (msg: any, sender: chrome.runtime.MessageSender, sendResponse: any) => {
      const messageResponse = callback(msg, sender, sendResponse);

      if (!messageResponse) {
        return false;
      }

      Promise.resolve(messageResponse);
      return true;
    }
  );
```

@differsthecat 

So I think it might make sense to treat this a bit more generally. What we should be checking within this helper is if the `callback` method returns any value at all. If a value is being returned, even if it isn't a promise, resolving the value and returning true will ensure that the receiving side is expecting a response with a value.

Otherwise, returning a `false` tells the message sender that no value should be waited for.